android/src/main/java/com/urbanairship/reactnative/ReactMessageViewManager.java

```diff
@@ -0,0 +1,57 @@
+/* Copyright Airship and Contributors */
+
+package com.urbanairship.reactnative;
+
+import androidx.annotation.Nullable;
+
+import com.facebook.react.common.MapBuilder;
+import com.facebook.react.uimanager.SimpleViewManager;
+import com.facebook.react.uimanager.ThemedReactContext;
+import com.facebook.react.uimanager.annotations.ReactProp;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+
+import javax.annotation.Nonnull;
+
+public class ReactMessageViewManager extends SimpleViewManager<ReactMessageView> {
+
+    public static final String REACT_CLASS = "UARCTMessageView";
+
+    @Override
+    public String getName() {
+        return REACT_CLASS;
+    }
+
+    @Nonnull
+    @Override
+    protected ReactMessageView createViewInstance(@Nonnull ThemedReactContext reactContext) {
+        return new ReactMessageView(reactContext);
+    }
+
+    @ReactProp(name = "messageId")
+    public void setMessageId(ReactMessageView view, @Nullable String messageId) {
+        view.loadMessage(messageId);
+    }
+
+    public Map getExportedCustomBubblingEventTypeConstants() {
```

Comment:
what is this section for? it looks like it has a fair amount of magic strings

Reply:
Really confused by this too. Before adding this code it was throwing an exception about an unexpected event. It seems react needs some sort of registry on Android - https://facebook.github.io/react-native/docs/native-components-android#events

Reply:
weird